backend/nodes/ncnn_nodes.py

```diff
@@ -189,18 +189,12 @@ def run(self, net_tuple: tuple, img: np.ndarray) -> np.ndarray:
         # Load model param and bin
         net.load_param(param_path)
 
-        bin_is_fp16 = True  # bin_data.dtype == "float16"
-
         with tempfile.TemporaryDirectory(prefix="chaiNNer-") as tempdir:
             bin_file_data = struct.pack(
                 "<I",
-                (
-                    FLAG_FLOAT_16
-                    if bin_is_fp16 or os.environ["isFp16"]
-                    else FLAG_FLOAT_32
-                ),
+                (FLAG_FLOAT_16 if os.environ["isFp16"] else FLAG_FLOAT_32),

```

Comment:
`os.environ` only contains strings, right? If so, then this won't work.

Also, could you please extract the `os.environ["isFp16"]` condition out into its own variable? I don't like that it's duplicated here and 2 lines below.

```py
>>> print(1 if False else 0)
0
>>> print(1 if "False" else 0)
1
```

Reply:
Looking at all the `os.environ["isFp16"]` in other places, maybe it would make sense to have 2 functions: 1one to read/parse the env into boolean, and one to set the env using boolean values.